### PR TITLE
[feature](iceberg)The new DDL syntax is added to create iceberg partitioned tables

### PR DESCRIPTION
**Pull request by wuwenchi**
## Proposed changes

Issue #31442

support partition by :

```
create table tb1 (c1 string, ts datetime) engine = iceberg partition by (c1, day(ts)) () properties ("a"="b")
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateTableInfo.java [hunk 1]
```diff
@@ -747,4 +747,8 @@ public CreateTableStmt translateToLegacyStmt() {
                 partitionDesc, distributionDesc, Maps.newHashMap(properties), extProperties,
                 comment, addRollups, null);
     }
+
+    public void setExternal() {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/info/CreateTableInfo.java:751]: this interface should follow standard setter. `setExternal(boolean isExternal)`

**Review by kaka11chen — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**